pkg/dmaasbackup/veleroschedules.go

```diff
@@ -32,24 +35,90 @@ func (o ScheduleByCreationTimestamp) Less(i, j int) bool {
 	return (*o[i].CreationTimestamp).Before(o[j].CreationTimestamp)
 }
 
-func getActiveVeleroSchedule(dbkp *v1alpha1.DMaaSBackup) *v1alpha1.VeleroScheduleDetails {
+// getLatestVeleroSchedule return the latest schedule from status.veleroschedule
+func getLatestVeleroSchedule(dbkp *v1alpha1.DMaaSBackup) *v1alpha1.VeleroScheduleDetails {
 	if len(dbkp.Status.VeleroSchedules) == 0 {
 		return nil
 	}
 
-	// sort veleroschedule with descending creationtimestamp
-	// veleroschedule have only one active schedule
-	sort.Sort(sort.Reverse(ScheduleByCreationTimestamp(dbkp.Status.VeleroSchedules)))
-
 	return &dbkp.Status.VeleroSchedules[0]
 }
 
+// getLastVeleroSchedule return the last created schedule from status.veleroschedule
+func getLastVeleroSchedule(dbkp *v1alpha1.DMaaSBackup) *v1alpha1.VeleroScheduleDetails {
+	if len(dbkp.Status.VeleroSchedules) < 2 {
+		return nil
+	}
+
+	return &dbkp.Status.VeleroSchedules[1]
+}
+
 // appendVeleroSchedule add given schedule to scheduledetails with Active status
+// and sort dbkp.Status.VeleroSchedules using schedule creationTimestamp in reverse order
 func appendVeleroSchedule(dbkp *v1alpha1.DMaaSBackup, schedule *velerov1api.Schedule) {
```

Comment:
this function doesn't seems to be in use

Reply:
yes